src/scripts/main.js

```diff
@@ -1,3 +1,41 @@
 'use strict';
 
-// write code here
+const empoyeeslist = document.querySelector('ul');
+
+function sortList(list) {
+  function moneyToNumber(string) {
+    const filtredString = string.replace(/\$/g, '').replace(/,/g, '');
+
+    return filtredString;
+  }
+
+  const listArray = [...list.querySelectorAll('li')];
+  const compareFunction = (a, b) =>
+    moneyToNumber(b.dataset.salary) - moneyToNumber(a.dataset.salary);
+  const sortedList = [...listArray].sort(compareFunction);
+
+  list.innerHTML = '';
+
+  for (const elem of sortedList) {
+    list.append(elem);
+  }
+}
+
+function getEmployees(list) {
+  const result = [];
+  const listArray = [...list.querySelectorAll('li')];
+
+  for (const elem of listArray) {
+    result.push({
+      name: elem.innerText,
+      position: elem.dataset.position,
+      salary: elem.dataset.salary,
+      age: elem.dataset.age,
+    });
+  }
+
+  return result;
+}
+
+sortList(empoyeeslist);
```

Comment:
Ensure that the `empoyeeslist` variable is correctly selecting the intended DOM element. If the element is not found, this will lead to errors.